lib/srv/desktop/rdp/rdpclient/Cargo.toml

```diff
@@ -5,7 +5,7 @@ authors = ["Andrew Lytvynov <andrew@goteleport.com>", "Zac Bergquist <zac@gotele
 edition = "2018"
 
 [lib]
-crate-type = ["staticlib"]
+crate-type = ["lib", "staticlib"]
```

Comment:
I've added a benchmark for encoding data in Rust and this change is needed to make a benchmark library (criterion) work. Without this change when we build our library it'll only produce the static library that we later link to the go binary, but the criterion can't reference to in the benchmarks. It needs a Rust library to compile benchmarks. Maybe there is another way to use only static library when doing benchmarks, but I don't know how this can be done.

Reply:
Does this increase the build times since we have to build 2 artifacts now?

I would be fine with just using the cargo benchmarking built in, even though it requires the latest nightly.

The benchmark made sense in Go because we made a few tweaks and wanted to see how it impacted performance. Honestly I think it's not as useful here in the Rust because

- we've already shown that its 3x faster
- there's not much more customization that can be done here

Reply:
The addition of the `lib` crate-type just slightly increase the build times. Our code base is small so it does not make much of a difference. In my local setup, for a full clean release build, it went from 128s to 129s.

I've added this benchmark as it made a sense to have some sort of a benchmark for encoding function in our Rust code especially that one was removed from the Go code. Another thing is that it already provides us a tools to create a different benchmarks for other part of the code if we ever want to measure performance.

I can remove it if you want as I agree that it does not give us much right now, but if we ever want to add benchmarking tool to our code base I think that criterion is the best choice.